src/Gui/MainWindow.cpp

```diff
@@ -1933,11 +1938,14 @@ void MainWindow::startSplasher()
 
 void MainWindow::stopSplasher()
 {
-    if (d->splashscreen) {
-        d->splashscreen->finish(this);
-        delete d->splashscreen;
-        d->splashscreen = nullptr;
-    }
+    QApplication::processEvents();
+    QTimer::singleShot(3000, this, [this]() {
+        if (d->splashscreen) {
+            d->splashscreen->finish(this);
+            d->splashscreen->deleteLater();
+            d->splashscreen = nullptr;
+        }
+    });
```

Comment:
This change makes splash to be displayed for a very short time. However it would be nice if someone with fast permanent storage (SSD, NVMe) could test this as well.

Reply:
@3x380V  I can add the wayland case here to to make it works as non-patched version. It is still very difficult to understand why closing the splash immediately takes more time than closing it 3 seconds later. But as you said, theory and practice are not the same.

Reply:
I guess the answer lays in `waitForWindowExposed()` implementation, but it is just a guess. With enough spare time it could probably be debugged using hardware timer, so we'd have more that a guesses.

Reply:
@3x380V Ok, added the conditional there too.